conda build undefined references

https://github.com/conda-forge/stir-feedstock/pull/147 fails on Linux builds with
```
: && $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/stir-6.3.0dev3 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/stir-6.3.0dev3 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix  -fno-sized-deallocation -msse2  -fno-sized-deallocation -msse2 -O3 -DNDEBUG -ffast-math -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib   -Wl,--dependency-file=src/analytic/DDSR2D/CMakeFiles/DDSR2D.dir/link.d src/CMakeFiles/stir_registries.dir/IO/IO_registries.cxx.o src/CMakeFiles/stir_registries.dir/buildblock/buildblock_registries.cxx.o src/CMakeFiles/stir_registries.dir/recon_buildblock/recon_buildblock_registries.cxx.o src/CMakeFiles/stir_registries.dir/data_buildblock/data_buildblock_registries.cxx.o src/CMakeFiles/stir_registries.dir/Shape_buildblock/Shape_buildblock_registries.cxx.o src/CMakeFiles/stir_registries.dir/modelling_buildblock/modelling_registries.cxx.o src/CMakeFiles/stir_registries.dir/spatial_transformation_buildblock/spatial_transformation_registries.cxx.o src/CMakeFiles/stir_registries.dir/scatter_buildblock/scatter_registries.cxx.o src/analytic/DDSR2D/CMakeFiles/DDSR2D.dir/DDSR2D.cxx.o -o src/analytic/DDSR2D/DDSR2D  src/IO/libIO.a  src/analytic/FBP3DRP/libanalytic_FBP3DRP.a  src/analytic/FBP2D/libanalytic_FBP2D.a  src/analytic/SRT2D/libanalytic_SRT2D.a  src/analytic/SRT2DSPECT/libanalytic_SRT2DSPECT.a  src/analytic/GRD2D/libanalytic_GRD2D.a  src/analytic/DDSR2D/libanalytic_DDSR2D.a  src/iterative/OSMAPOSL/libiterative_OSMAPOSL.a  src/iterative/KOSMAPOSL/libiterative_KOSMAPOSL.a  src/iterative/OSSPS/libiterative_OSSPS.a  src/scatter_buildblock/libscatter_buildblock.a  src/modelling_buildblock/libmodelling_buildblock.a  src/listmode_buildblock/liblistmode_buildblock.a  src/recon_buildblock/librecon_buildblock.a  src/display/libdisplay.a  src/data_buildblock/libdata_buildblock.a  src/numerics_buildblock/libnumerics_buildblock.a  src/buildblock/libbuildblock.a  src/spatial_transformation_buildblock/libspatial_transformation_buildblock.a  src/Shape_buildblock/libShape_buildblock.a  src/eval_buildblock/libeval_buildblock.a  $PREFIX/lib/libparallelproj_c.so.1.10.2  src/IO/libIO.a  src/modelling_buildblock/libmodelling_buildblock.a  src/IO/libIO.a  src/modelling_buildblock/libmodelling_buildblock.a  src/listmode_buildblock/liblistmode_buildblock.a  src/data_buildblock/libdata_buildblock.a  $PREFIX/lib/libTree.so.6.36.04  $PREFIX/lib/libMathCore.so.6.36.04  $PREFIX/lib/libImt.so.6.36.04  $PREFIX/lib/libMultiProc.so.6.36.04  $PREFIX/lib/libNet.so.6.36.04  $PREFIX/lib/libRIO.so.6.36.04  $PREFIX/lib/libThread.so.6.36.04  $PREFIX/lib/libCore.so.6.36.04  $PREFIX/lib/libhdf5_cpp.so  $PREFIX/lib/libhdf5.so  $PREFIX/lib/libcrypto.so  $PREFIX/lib/libcurl.so  $PREFIX/lib/libsz.so  $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib/libdl.so  $BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot/usr/lib/libm.so  $PREFIX/lib/libITKConvolution-5.4.so.1  $PREFIX/lib/libitkAdaptiveDenoising-5.4.so.1  $PREFIX/lib/libITKColormap-5.4.so.1  $PREFIX/lib/libITKDICOMParser-5.4.so.1  $PREFIX/lib/libITKDenoising-5.4.so.1  $PREFIX/lib/libITKIOBruker-5.4.so.1  $PREFIX/lib/libITKIOCSV-5.4.so.1  $PREFIX/lib/libITKIOHDF5-5.4.so.1  $PREFIX/lib/libjpeg.so  $PREFIX/lib/libITKIOJPEG2000-5.4.so.1  $PREFIX/lib/libtiff.so  $PREFIX/lib/libITKIOLSM-5.4.so.1  $PREFIX/lib/libITKIOMRC-5.4.so.1  $PREFIX/lib/libpng.so  $PREFIX/lib/libITKIOTransformMINC-5.4.so.1  $PREFIX/lib/libITKIOMINC-5.4.so.1  $PREFIX/lib/libitkminc2-5.4.so.1  $PREFIX/lib/libhdf5_cpp.so  $PREFIX/lib/libhdf5.so  $PREFIX/lib/libhdf5_hl.so  -lrt  $PREFIX/lib/libITKRegistrationMethodsv4-5.4.so.1  $PREFIX/lib/libITKOptimizersv4-5.4.so.1  $PREFIX/lib/libitklbfgs-5.4.so.1  $PREFIX/lib/libITKReview-5.4.so.1  $PREFIX/lib/libITKLabelMap-5.4.so.1  $PREFIX/lib/libITKFastMarching-5.4.so.1  $PREFIX/lib/libITKPolynomials-5.4.so.1  $PREFIX/lib/libITKBiasCorrection-5.4.so.1  $PREFIX/lib/libITKDeformableMesh-5.4.so.1  $PREFIX/lib/libITKDiffusionTensorImage-5.4.so.1  $PREFIX/lib/libITKPDEDeformableRegistration-5.4.so.1  $PREFIX/lib/libITKOptimizers-5.4.so.1  $PREFIX/lib/libITKIOBioRad-5.4.so.1  $PREFIX/lib/libITKIOGE-5.4.so.1  $PREFIX/lib/libitkopenjpeg-5.4.so.1  $PREFIX/lib/libITKIOSiemens-5.4.so.1  $PREFIX/lib/libITKIOIPL-5.4.so.1  $PREFIX/lib/libITKIOSpatialObjects-5.4.so.1  $PREFIX/lib/libITKIOXML-5.4.so.1  $PREFIX/lib/libITKIOStimulate-5.4.so.1  $PREFIX/lib/libITKIOTransformHDF5-5.4.so.1  $PREFIX/lib/libITKIOTransformInsightLegacy-5.4.so.1  $PREFIX/lib/libITKIOTransformMatlab-5.4.so.1  $PREFIX/lib/libITKIOTransformBase-5.4.so.1  $PREFIX/lib/libITKTransformFactory-5.4.so.1  $PREFIX/lib/libITKKLMRegionGrowing-5.4.so.1  $PREFIX/lib/libITKMarkovRandomFieldsClassifiers-5.4.so.1  $PREFIX/lib/libITKQuadEdgeMeshFiltering-5.4.so.1  $PREFIX/lib/libITKRegionGrowing-5.4.so.1  $PREFIX/lib/libITKVTK-5.4.so.1  $PREFIX/lib/libITKWatersheds-5.4.so.1  $PREFIX/lib/libITKMathematicalMorphology-5.4.so.1  $PREFIX/lib/libITKStatistics-5.4.so.1  $PREFIX/lib/libitkNetlibSlatec-5.4.so.1  $PREFIX/lib/libITKPath-5.4.so.1  $PREFIX/lib/libITKTestKernel-5.4.so.1  $PREFIX/lib/libITKFFT-5.4.so.1  -lfftw3_threads  -lfftw3  -lfftw3f_threads  -lfftw3f  $PREFIX/lib/libITKIOBMP-5.4.so.1  $PREFIX/lib/libITKIOGDCM-5.4.so.1  $PREFIX/lib/libitkgdcmMSFF-5.4.so.1  $PREFIX/lib/libitkgdcmDICT-5.4.so.1  $PREFIX/lib/libitkgdcmIOD-5.4.so.1  $PREFIX/lib/libitkgdcmDSED-5.4.so.1  $PREFIX/lib/libitkgdcmCommon-5.4.so.1  $PREFIX/lib/libITKIOGIPL-5.4.so.1  $PREFIX/lib/libITKIOJPEG-5.4.so.1  $PREFIX/lib/libITKIOTIFF-5.4.so.1  $PREFIX/lib/libITKIOMeshBYU-5.4.so.1  $PREFIX/lib/libITKIOMeshFreeSurfer-5.4.so.1  $PREFIX/lib/libITKIOMeshGifti-5.4.so.1  $PREFIX/lib/libITKgiftiio-5.4.so.1  $PREFIX/lib/libexpat.so  $PREFIX/lib/libITKIOMeshOBJ-5.4.so.1  $PREFIX/lib/libITKIOMeshOFF-5.4.so.1  $PREFIX/lib/libITKIOMeshVTK-5.4.so.1  $PREFIX/lib/libITKIOMeshBase-5.4.so.1  $PREFIX/lib/libITKQuadEdgeMesh-5.4.so.1  $PREFIX/lib/libITKIOMeta-5.4.so.1  $PREFIX/lib/libITKMetaIO-5.4.so.1  $PREFIX/lib/libITKIONIFTI-5.4.so.1  $PREFIX/lib/libITKniftiio-5.4.so.1  $PREFIX/lib/libITKznz-5.4.so.1  $PREFIX/lib/libITKIONRRD-5.4.so.1  $PREFIX/lib/libITKNrrdIO-5.4.so.1  $PREFIX/lib/libITKIOPNG-5.4.so.1  $PREFIX/lib/libITKIOVTK-5.4.so.1  $PREFIX/lib/libITKVideoIO-5.4.so.1  $PREFIX/lib/libITKVideoCore-5.4.so.1  $PREFIX/lib/libitkMGHIO-5.4.so.1  $PREFIX/lib/libz.so  $PREFIX/lib/libITKIOImageBase-5.4.so.1  $PREFIX/lib/libitkSimpleITKFilters-5.4.so.1  $PREFIX/lib/libITKImageFeature-5.4.so.1  $PREFIX/lib/libITKSpatialObjects-5.4.so.1  $PREFIX/lib/libITKMesh-5.4.so.1  $PREFIX/lib/libITKTransform-5.4.so.1  $PREFIX/lib/libITKCommon-5.4.so.1  $PREFIX/lib/libitkdouble-conversion-5.4.so.1  $PREFIX/lib/libitksys-5.4.so.1  $PREFIX/lib/libITKVNLInstantiation-5.4.so.1  $PREFIX/lib/libitkvnl_algo-5.4.so.1  $PREFIX/lib/libitkvnl-5.4.so.1  $PREFIX/lib/libitkv3p_netlib-5.4.so.1  $PREFIX/lib/libitkvcl-5.4.so.1  -lm  -lpthread  -lm  -ldl  $PREFIX/lib/libITKSmoothing-5.4.so.1  src/numerics_buildblock/libnumerics_buildblock.a  src/buildblock/libbuildblock.a  $PREFIX/lib/libgomp.so  -lpthread && :
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: src/analytic/DDSR2D/libanalytic_DDSR2D.a(DDSR2DReconstruction.cxx.o): in function `stir::DDSR2DReconstruction::actual_reconstruct(std::shared_ptr<stir::DiscretisedDensity<3, float> > const&)':
DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x2986): undefined reference to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)'
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x29c2): undefined reference to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)'
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x2e9e): undefined reference to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)'
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x2ed8): undefined reference to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)'
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x323e): undefined reference to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)'
$BUILD_PREFIX/bin/../lib/gcc/x86_64-conda-linux-gnu/14.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: src/analytic/DDSR2D/libanalytic_DDSR2D.a(DDSR2DReconstruction.cxx.o):DDSR2DReconstruction.cxx:(.text._ZN4stir20DDSR2DReconstruction18actual_reconstructERKSt10shared_ptrINS_18DiscretisedDensityILi3EfEEE+0x3278): more undefined references to `void stir::fourier_1d<stir::Array<1, std::complex<float> > >(stir::Array<1, std::complex<float> >&, int)' follow
```

--- a/src/numerics_buildblock/fourier.cxx
+++ b/src/numerics_buildblock/fourier.cxx
@@ -387,6 +387,7 @@ template void fourier<>(Array<1, std::complex<float>>& c, const int sign);
 template void fourier<>(VectorWithOffset<std::complex<float>>& c, const int sign);
 
 #define INSTANTIATE(d, type)                                                                                                     \
+  template void fourier_1d<>(Array<d, std::complex<type>> & v, const int sign);                                                  \
   template Array<d, std::complex<type>> fourier_for_real_data<>(const Array<d, type>& v, const int sign);                        \
   template Array<d, type> inverse_fourier_for_real_data_corrupting_input<>(Array<d, std::complex<type>> & c, const int sign);    \
   template Array<d, type> inverse_fourier_for_real_data<>(const Array<d, std::complex<type>>& c, const int sign);                \

